bodo/pandas/base.py

```diff
@@ -4,24 +4,35 @@
 from bodo.ext import plan_optimizer
 from bodo.pandas.frame import BodoDataFrame
 from bodo.pandas.series import BodoSeries
+from bodo.pandas.utils import LazyPlan, scatter_data
 
 
 def from_pandas(df):
     """Convert a Pandas DataFrame to a BodoDataFrame."""
     import pyarrow as pa
 
+    import bodo
+
     if not isinstance(df, pd.DataFrame):
         raise TypeError("Input must be a pandas DataFrame")
 
+    # TODO: Add support for Index
+    if (
+        not isinstance(df.index, pd.RangeIndex)
+        or df.index.start != 0
+        or df.index.step != 1
+    ):
+        raise ValueError("Only RangeIndex with start=0 and step=1 is supported")
```

Comment:
Created an issue since we need to add Index handling in general to our tables similar to Arrow probably. Parquet read development is probably a good time to do it.